Description: Use absolute imports to support building with gccgo.
Author: Dimitri John Ledkov <xnox@ubuntu.com>

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@
 STAGEDIR = $(BUILDDIR)/stage
 OBJECTDIR = $(BUILDDIR)/obj
 DISTDIR = $(BUILDDIR)/dist
+GOSRCDIR = $(BUILDDIR)/src/xe-guest-utilities
 
 OBJECTS :=
 OBJECTS += $(OBJECTDIR)/xe-daemon
@@ -21,7 +22,7 @@
 PACKAGE = xe-guest-utilities
 VERSION = $(PRODUCT_VERSION)
 RELEASE := $(shell git rev-list HEAD | wc -l)
-ARCH := $(shell go version|awk -F'/' '{print $$2}')
+ARCH := $(shell go version|awk -F'/' '{print $$NF}')
 
 ifeq ($(ARCH), amd64)
 	ARCH = x86_64
@@ -65,13 +66,17 @@
 	  tar cf $@ * \
 	)
 
-$(OBJECTDIR)/xe-daemon: $(XE_DAEMON_SOURCES:%=$(GOBUILDDIR)/%)
+$(GOSRCDIR):
+	mkdir -p $(BUILDDIR)/src/
+	ln -sf $(GOBUILDDIR) $@
+
+$(OBJECTDIR)/xe-daemon: $(XE_DAEMON_SOURCES:%=$(GOBUILDDIR)/%) $(GOSRCDIR)
 	mkdir -p $(OBJECTDIR)
-	$(GO_BUILD) $(GO_FLAGS) -o $@ $<
+	GOPATH=$(BUILDDIR) $(GO_BUILD) $(GO_FLAGS) -o $@ xe-guest-utilities/xe-daemon
 
 $(OBJECTDIR)/xenstore: $(XENSTORE_SOURCES:%=$(GOBUILDDIR)/%) $(GOROOT)
 	mkdir -p $(OBJECTDIR)
-	$(GO_BUILD) $(GO_FLAGS) -o $@ $<
+	GOPATH=$(BUILDDIR) $(GO_BUILD) $(GO_FLAGS) -o $@ xe-guest-utilities/xenstore
 
 $(GOBUILDDIR)/%: $(REPO)/%
 	mkdir -p $$(dirname $@)
--- a/guestmetric/guestmetric_linux.go
+++ b/guestmetric/guestmetric_linux.go
@@ -1,7 +1,7 @@
 package guestmetric
 
 import (
-	xenstoreclient "../xenstoreclient"
+	"xe-guest-utilities/xenstoreclient"
 	"bufio"
 	"bytes"
 	"fmt"
--- a/xe-daemon/xe-daemon.go
+++ b/xe-daemon/xe-daemon.go
@@ -1,10 +1,10 @@
 package main
 
 import (
-	guestmetric "../guestmetric"
-	syslog "../syslog"
-	system "../system"
-	xenstoreclient "../xenstoreclient"
+	"xe-guest-utilities/guestmetric"
+	"xe-guest-utilities/syslog"
+	system "xe-guest-utilities/system"
+	"xe-guest-utilities/xenstoreclient"
 	"flag"
 	"fmt"
 	"io"
--- a/xenstore/xenstore.go
+++ b/xenstore/xenstore.go
@@ -1,7 +1,7 @@
 package main
 
 import (
-	xenstoreclient "../xenstoreclient"
+	"xe-guest-utilities/xenstoreclient"
 	"fmt"
 	"os"
 	"strings"
--- a/xenstoreclient/xenstore.go
+++ b/xenstoreclient/xenstore.go
@@ -1,7 +1,7 @@
 package xenstoreclient
 
 import (
-	syslog "../syslog"
+	"xe-guest-utilities/syslog"
 	"bufio"
 	"bytes"
 	"encoding/binary"
